.github/workflows/test-code.yml

```diff
@@ -76,6 +76,11 @@ jobs:
         if: ${{ startsWith(matrix.os, 'windows') }}
         working-directory: src
         run: dotnet test --no-restore --no-build
+      - name: Test AOT compilation
+        working-directory: src/GraphQL.AotCompilationSample
+        run: |
+          dotnet publish -c Release -o publish
+          publish/GraphQL.AotCompilationSample
```

Comment:
This will fail the build check if the application (a) returns a non-zero exit code, or (b) throws an exception which returns a non-zero exit code.

Reply:
👍   I would change "Test AOT compilation" to something like "Test publishing with AOT". As I understand AOT phase works after dotnet build. It's a sort of "post-compilation".

Reply:
I see no RuntimeIdentifier either for dotnet publish CLI or in csproj file. So RuntimeIdentifier here is inferred automatically - https://learn.microsoft.com/en-us/dotnet/core/compatibility/sdk/7.0/automatic-runtimeidentifier?source=recommendations + https://learn.microsoft.com/en-us/dotnet/core/compatibility/sdk/7.0/automatic-rid-publish-only . Maybe to add a note about that behavior in example to not confuse AOT newcomers?

Reply:
From https://learn.microsoft.com/en-us/dotnet/core/deploying/trimming/trimming-options?pivots=dotnet-7-0 : 
> Some of the options mention ILLink, which is the name of the underlying tool that implements trimming.